pkg/gatherers/conditional/gather_containers_logs.go

```diff
@@ -57,44 +63,38 @@ func (g *Gatherer) gatherContainersLogs(
 	var records []record.Record
 
 	for _, alertLabels := range alertInstances {
-		podNamespace, err := getAlertPodNamespace(alertLabels)
+		info, err := parseAlertLabels(alertLabels)
 		if err != nil {
 			klog.Warningf(logMissingAlert, err.Error(), params.AlertName)
 			errs = append(errs, err)
-			continue
 		}
-		podName, err := getAlertPodName(alertLabels)
-		if err != nil {
-			klog.Warningf(logMissingAlert, err.Error(), params.AlertName)
-			errs = append(errs, err)
+		if len(info.namespace) == 0 {
 			continue
 		}
-		var podContainer string
-		if len(params.Container) > 0 {
-			podContainer = params.Container
-		} else {
-			podContainer, err = getAlertPodContainer(alertLabels)
-			if err != nil {
-				klog.Warningf(logMissingAlert, err.Error(), params.AlertName)
-				errs = append(errs, err)
-			}
+
+		logContainersFilter := &common.LogContainersFilter{
+			Namespace: info.namespace,
 		}
 
-		logContainersFilter := common.LogContainersFilter{
-			Namespace:     podNamespace,
-			FieldSelector: fmt.Sprintf("metadata.name=%s", podName),
+		if len(params.Container) > 0 {
+			logContainersFilter.ContainerNameRegexFilter = fmt.Sprintf("^%s$", params.Container)
+		} else if len(info.container) > 0 {
+			// KubePod* conditions does not have the container so, we need to verify it first,
+			// and then we can apply it to the creation of the regex rule.
+			logContainersFilter.ContainerNameRegexFilter = fmt.Sprintf("^%s$", info.container)
```

Comment:
I am afraid that this will lead to gathering of logs from the whole namespace for `KubePod*` alerts, because the `ContainerNameRegexFilter` as well as `PodNameRegexFilter` will not be set in this case. Can you please verify it?

Reply:
Just to keep history from our discussion:

This will not happen because of the alert. KubePod* will always have the PodName on its Alert given the nature of the alert itself.

```json
{
  "metric": {
    "__name__": "ALERTS",
    "alertname": "KubePodNotReady",
    "alertstate": "firing",
    "namespace": "openshift-storage",
    "pod": "noobaa-core-0",
    "severity": "warning"
  },
  "value": [
    1674040588.66,
    "1"
  ]
}
{
  "metric": {
    "__name__": "ALERTS",
    "alertname": "KubePodNotReady",
    "alertstate": "firing",
    "namespace": "openshift-storage",
    "pod": "noobaa-db-pg-0",
    "severity": "warning"
  },
  "value": [
    1674040588.66,
    "1"
  ]
}
```

So, this is not a problem... also, if I check for PodName exists, it will remove the possibility of getting logs from containers inside a namespace. And If remember correctly, and I can test it, it will impact the `AlertmanagerFailedToSendAlerts`